### PR TITLE
feat: replace parsePackageAuthor with npm package

### LINKED ISSUE #789: 🛠 Tooling: Replace parsePackageAuthor with npm package
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/create-typescript-app/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

This `parsePackageAuthor` function exists in the repository:

https://github.com/JoshuaKGoldberg/create-typescript-app/blob/a386c870f5bc53ce3b30a8d31108c5726487ba78/src/shared/options/readGitAndNpmDefaults/parsePackageAuthor.ts#L3

...but https://www.npmjs.com/package/parse-author already exists. There's no need for this repo to implement that logic when it could use an npm package.

Let's delete the `parsePackageAuthor` function (and its tests) from this repo, then use the `parse-author` package instead.

### Additional Info

_No response_

### PR DESCRIPTION
**Pull request by kristo-baricevic**
## PR Checklist

- [x] Addresses an existing open issue: fixes #789
- [x] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/create-typescript-app/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/create-typescript-app/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

#789 parse-author dependency was installed and the parse method was used to replace split and trim logic on the string containing author's name and email

### CODE REVIEW

**Comment by kristo-baricevic** [comment]
I was working through the checks git showed me, and after running pnpm lint the checks back been stuck on "expected - waiting for status to be reported" 

Let me know how the titling and additional info looks! I enjoyed working on this feature.

**Comment by JoshuaKGoldberg** [comment]
We chatted 1:1 a bit - the merge conflicts are stopping builds from running. And also looks like there are some other changes in the mix.

**Comment by kristo-baricevic** [comment]
I am not exactly sure what happened there but I do know it happened when I was trying to respond to the red X's in git github checks, and it involved some installs and running npx create-typescript-app. So I backed up the HEAD of the branch to the commit before those changes were made.

File: src/shared/options/readOptionDefaults/parsePackageAuthor.ts [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
 import { PartialPackageData } from "../../types.js";
 
 export function parsePackageAuthor(packageData: PartialPackageData) {
 	const [packageAuthor, packageEmail] =
 		typeof packageData.author === "string"
-			? [
-					packageData.author.split("<")[0].trim(),
-					packageData.author.split(/<|>/)[1]?.trim(),
-			  ]
+			? [parse(packageData.author).name, parse(packageData.author).email]
```
Comment (JoshuaKGoldberg) [inline src/shared/options/readOptionDefaults/parsePackageAuthor.ts:8]: Good direction! I'd rather this only `parse` once though. Might have to start putting an `if` statement or two in 😄

Comment (kristo-baricevic) [reply src/shared/options/readOptionDefaults/parsePackageAuthor.ts:8]: Ok thanks for the feedback - I'll look into this!

Comment (kristo-baricevic) [reply src/shared/options/readOptionDefaults/parsePackageAuthor.ts]: Hey Josh - I reconfigured the code to include an if statement and a variable to hold the parsed author information. Let me know if you like how it looks.

**Review by JoshuaKGoldberg — APPROVED** [review]
LGTM, thanks! Clean implementation and passes all the existing tests. Nice.

Once you resolve the merge conflicts we should be good to go. 🚀

**Review by JoshuaKGoldberg — APPROVED** [review]
💯 swell, thanks @kristo-baricevic!

**Comment by JoshuaKGoldberg** [comment]
Initialization script failures are unrelated.

**Pull request merged**